src/main/java/io/kestra/plugin/gcp/vertexai/ChatCompletion.java

```diff
@@ -0,0 +1,140 @@
+package io.kestra.plugin.gcp.vertexai;
+
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.runners.RunContext;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.MutableHttpRequest;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+
+import java.util.List;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+
+import static io.kestra.core.utils.Rethrow.throwFunction;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Chat completion using the Vertex AI PaLM API for Google's PaLM 2 large language models (LLM)",
+    description = "See [Generative AI quickstart using the Vertex AI API](https://cloud.google.com/vertex-ai/docs/generative-ai/start/quickstarts/api-quickstart) for more information."
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Chat completion using the Vertex AI PaLM API",
+            code = {
+                """
+                    region: us-central1
+                    projectId: my-project
+                    context: I love jokes that talk about sport
+                    messages:
+                      - author: user
+                        content: Please tell me a joke"""
```

Comment:
```suggestion
                        content: tell me a joke"""
```

you're so polite 😄

Reply:
Well, it's a habit I took always to be polite with bots, you never know ... Skynet...